### DISCUSSION TITLE
How to make it work for 96*48 1/24 scan rgb matrix

### DISCUSSION THREAD

**Discussion by VijayVinayaraj** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#550] [Q&A] [answered]
I bought a 96*48 P2.5 1/24 scan RGB matrix from Waveshare. Their official documentation suggests to use your library. But I couldn't find any documentation on support for 1/24 scan rate for matrix.
https://www.waveshare.com/wiki/RGB-Matrix-P2.5-96x48-F
This is the doc of Waveshare.

**Comment by board707** [comment]
Please indicate the partnumbers of the chips on the rear side

**Reply by VijayVinayaraj** [reply]
NCSD-p2.5-96*48(2121)-24S-V2.1
ICN203778p
SM5166PF
MW245B

**Reply by board707** [reply]
All drivers are standard. Scan 24 at height 48 - nothing special.
The panel should be supported by library.

**Reply by VijayVinayaraj** [reply]
This is what I get When I run this code . Every colour is divided like this. 
Could you please help me with this?


```
#include <Arduino.h>
#include "ESP32-HUB75-MatrixPanel-I2S-DMA.h"

MatrixPanel_I2S_DMA *dma_display = nullptr;
uint16_t myRED = dma_display->color565(255, 0, 0);
uint16_t myGREEN = dma_display->color565(150, 150, 100);
uint16_t myBLUE = dma_display->color444(15, 0, 15);
void setup() {
  HUB75_I2S_CFG mxconfig(/* width = */ 96, /* height = */ 48, /* chain = */ 1);
mxconfig.gpio.e=32;
  dma_display = new MatrixPanel_I2S_DMA(mxconfig);
  dma_display->begin();
  dma_display->setBrightness8(80);
  dma_display->clearScreen();
  // `println` is only available when the Adafruit GFX library is used.

}
void loop(){
dma_display->fillScreen(myBLUE);
  delay(2000);
  dma_display->fillScreen(myRED);
  delay(2000);
  dma_display->fillScreen(myGREEN);
  delay(2000);
}

```


![image](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/24764527/f9eae27b-473c-485e-a148-0a556080211d)

**Comment by VijayVinayaraj** [comment, marked as answer]
@board707  I found the issue in [common issues page](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/issues/134). I didn't see the page at first. The problem was GPIO pin for the address line. I changed the C,D pins from default pins to 21,19 respectively.
```
mxconfig.gpio.c = 21 ;
  mxconfig.gpio.d = 19;
```
This fixed the issue 
Thank you for the help. I thought this library didn't support the panel as there was no mention of 1/24 scan rate or the panel was defective.

**Reply by mrcodetastic** [reply]
Any ideas why changing the pins fixes this issue?

**Reply by board707** [reply]
Judging by the picture, only one of the two pins did not work  - most likely C. It is possible that the pin is poorly soldered on the ESP32 or an unstable connection in the wires. When @VijayVinayaraj changed the pin, the problem was gone

**Reply by board707** [reply]
As a test, OP can now try to return the default pins - by one at a time. First, change only D to the default one and try, and then only C. If in both cases the operation of the panel is not disrupted, switch both pins

**Reply by VijayVinayaraj** [reply]
Sure @board707, I will test that and get back to you.
